src/arch/arm/machine/hardware.c

```diff
@@ -32,17 +32,15 @@ BOOT_CODE const p_region_t *get_avail_p_regs(void)
 BOOT_CODE void map_kernel_devices(void)
 {
     for (int i = 0; i < ARRAY_SIZE(kernel_devices); i++) {
-        map_kernel_frame(kernel_devices[i].paddr,
-                         kernel_devices[i].pptr,
-                         VMKernelOnly,
-                         vm_attributes_new(kernel_devices[i].armExecuteNever,
-                                           false, false));
-        if (!kernel_devices[i].userAvailable) {
-            p_region_t reg = {
-                .start = kernel_devices[i].paddr,
-                .end = kernel_devices[i].paddr + BIT(PAGE_BITS),
-            };
-            reserve_region(reg);
+        const kernel_frame_t *frame = &kernel_device_regions[i];
```

Comment:
this commit also seems like needless churn. A code review and a commit to save 2 lines of code by using an alias?

Reply:
It's not about saving 2 lines of code, and I also have string faith in the compiler to be smart enough to detect that it can cache the pointer instead of calculating is everytime. This is about removing redundancy from the code by using a helper alias variable, because the array element is used multiple times. That make the code easier to understand as is separates what is done with the array element from the fact that this is an element from an array.
I also thing somebody mentioned that this pattern makes things a bit easier for verification, but I might be wrong here if have ripped this out of the original context.

Reply:
I'm not convinced optimising boot code is worth the review time, nor the very minor readability improvement. My main point is there's a cost to every PR line of code, so if we can reduce the unnecessary changes that would reduce reviewer load and also get your changes in faster. However it's done so I'll approve it. Please be aware changes like this reduce the reviewer bandwidth so other more important changes are delayed.